Content.Shared/Weapons/Hitscan/Systems/HitscanBasicVisualsSystem.cs

```diff
@@ -0,0 +1,86 @@
+using System.Numerics;
+using Content.Shared.Weapons.Hitscan.Components;
+using Content.Shared.Weapons.Hitscan.Events;
+using Content.Shared.Weapons.Ranged.Systems;
+using Robust.Shared.Map;
+using Robust.Shared.Player;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.Weapons.Hitscan.Systems;
+
+public sealed class HitscanBasicVisualsSystem : EntitySystem
+{
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<HitscanBasicVisualsComponent, HitscanRaycastFiredEvent>(OnHitscanHit, before: [ typeof(HitscanReflectSystem) ]);
+    }
+
+    private void OnHitscanHit(Entity<HitscanBasicVisualsComponent> hitscan, ref HitscanRaycastFiredEvent args)
+    {
+        FireEffects(args.FromCoordinates, args.DistanceTried, args.ShotDirection.ToAngle(), hitscan.Comp);
```

Comment:
Actually could arguably inline this method but I think it's fine. Just wanted to point it out.